### PR TITLE
[FIX] mrp: correct amounts in Consumption Warning

### PR DESCRIPTION
**Pull request by Sehnde**
Currently in 17.0, when adding an additional component on a Manufacturing Order, the Consumption Warning wizard shows:
- to consume: amount of additional component
- consumed: 0.0 

It should be the reverse. Moreover, this makes the 'Set Quantities & Validate' button have the same result as the 'Confirm' button, which is useless.

task 3670356

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/mrp/tests/test_order.py [hunk 1]
```diff
@@ -860,6 +860,43 @@ def test_consumption_warning_1(self):
         warning.action_confirm()
         self.assertEqual(mo.state, 'done')
 
+    def test_consumption_warning_2(self):
+        """ Test that additional components are correctly computed in the Consumption
+        Warning wizard and that the 'Set Quantities & Validate' works as intended.
+        Test that removing a bom component is also correctly computed in the warning.
```
Comment (ticodoo) [inline addons/mrp/tests/test_order.py:866]: :thinking: I already added in a test case for a deleted bom component into `test_consumption_action_set_qty_and_validate` and I think your fix is unrelated to it. I would say either leave it out of this test to make it easier to understand and/or add in some messages into your `assertEqual` to say what the expected behavior is.

**Review by ticodoo — COMMENTED** [review]
Just a small test comment.

Also, I would reword this part of your commit:

```
Moreover, this makes the 'Set Quantities & Validate'
button have the same result as the 'Confirm' button, which is useless.
```

 to make it clearer:

```
Additionally, this reversal makes the 'Set Quantities & Validate'
button have the same result as the 'Confirm' button for the additional component lines.
```

Also if you want to be completely precise, you should include in your commit message that the additional component is NOT in the BoM since the behavior is different if you add a new line for a component that is in the BoM.

File: addons/mrp/tests/test_order.py [hunk 2]
```diff
@@ -3511,47 +3512,58 @@ def test_consumption_action_set_qty_and_validate(self):
...
         self.assertEqual(mo2.move_raw_ids[0].quantity, 48, "current MO Consumed qty should match expected qty to produce based on manually set value")
+        self.assertEqual(mo2.move_raw_ids[1].product_uom_qty, 50, "current MO To Consume qty should match manually set expected qty produced")
+        self.assertEqual(mo2.move_raw_ids[1].quantity, 40, "current MO Consumed qty should match expected qty to produce based on manually set value")
 
         action = mo2.button_mark_done()
         warning = Form(self.env['mrp.consumption.warning'].with_context(**action['context']))
         consumption = warning.save()
-        self.assertEqual(len(consumption.mrp_consumption_warning_line_ids), 2, "deleted move should also show as an consumption line diff from BoM")
+        self.assertEqual(len(consumption.mrp_consumption_warning_line_ids), 3, "deleted move should also show as an consumption line diff from BoM")
         # mrp_consumption_warning_line_ids[1] = p1 => 12 unit qty_base, mrp_consumption_warning_line_ids[0] = p2 => 10 unit qty_base
```
Comment (ticodoo) [inline addons/mrp/tests/test_order.py:3539]:
```suggestion
        # mrp_consumption_warning_line_ids[2] = p1 => 12 unit qty_base
        # mrp_consumption_warning_line_ids[1] = p2 => 10 unit qty_base
        # mrp_consumption_warning_line_ids[0] = self.product_1 => 10 unit qty_base
```

**Review by ticodoo — COMMENTED** [review]
also don't forget to update the
`#### scenario 2 - that removing a line in the MO + changing the uom of a line ####`
line to match your changes

Otherwise good for me

@robodoo delegate=@Sehnde

**Comment by Sehnde** [comment]
@robodoo r+
